Fitness/Views/Cells/HabitTrackerCheckinCell.swift

```diff
@@ -0,0 +1,173 @@
+            make.trailing.equalToSuperview().offset(-32)
+            make.height.equalTo(22)
+            make.centerY.equalToSuperview()
+            make.width.equalTo(36)
+        }
+
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalTo(checkinCircle.snp.trailing).offset(11).priority(.high)
+            make.centerY.equalToSuperview()
+            make.height.equalTo(22)
+            make.trailing.lessThanOrEqualTo(streakLabel.snp.leading).offset(-2)
+
+        }
+        
+        separator.snp.makeConstraints { make in
+            make.height.equalTo(1)
+            make.leading.equalToSuperview().offset(16)
+            make.trailing.equalToSuperview().offset(-32).priority(.high)
+            make.bottom.equalToSuperview()
+        }
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        titleLabel.textColor = .fitnessBlack
+        checkinCircle.isSelected = false
+    }
+    
+    func configure(habit: Habit) {
+        self.habit = habit
+        
+        titleLabel.text = habit.title
+        streakLabel.text = getStreakEmoji()
+        
+        if let mostRecentCheckin = habit.dates.sorted().last {
+            if Date() - 86400.0 < mostRecentCheckin {
+                checkinCircle.isSelected = true
+                
+                let attributeString: NSMutableAttributedString =  NSMutableAttributedString(string: habit.title)
+                attributeString.addAttribute(NSAttributedString.Key.strikethroughStyle, value: 2, range: NSMakeRange(0, attributeString.length))
+                titleLabel.attributedText = attributeString
+                titleLabel.textColor = .fitnessMediumGrey
+            }
+        }
+    }
+    
+    @objc private func checkIn() {
+        switch checkinCircle.isSelected {
+        case true:
+            Habit.removeDate(habit: habit, date: Date())
+            habit.dates.removeFirst()
+            
+            titleLabel.attributedText = nil
+            titleLabel.text = habit.title
+            titleLabel.textColor = .fitnessBlack
+        case false:
+            Habit.logDate(habit: habit, date: Date())
+            
+            let attributeString: NSMutableAttributedString =  NSMutableAttributedString(string: habit.title)
+            attributeString.addAttribute(NSAttributedString.Key.strikethroughStyle, value: 2, range: NSMakeRange(0, attributeString.length))
+            titleLabel.attributedText = attributeString
+            titleLabel.textColor = .fitnessMediumGrey
+        }
+        
+        habit = Habit.getHabit(habit: habit.title, type: habit.type)
+        
+        checkinCircle.isSelected.toggle()
+        streakLabel.text = getStreakEmoji()
+    }
+    
+    private func getStreakEmoji() -> String {
+        let streak = habit.streak
+        
+        if streak <= 1 {
+            return ""
+        }
+        
+        let level: Int
+        if streak < 3 {
```

Comment:
Can we make this use a static constant dictionary that we loop over and keep track of the max level

Reply:
yup that's sleek!